blivet/osinstall.py

```diff
@@ -1297,6 +1297,37 @@ def get_fstype(self, mountpoint=None):
 
         return fstype
 
+    def sort_disks(self, req_disks):
+        """
+        Sort the disks to be used in parittioning, making sure the boot disk is
+        first.
+
+        :param req_disks: a list of required disks for partitioning
+        :type req_disks: list
+        :returns: The list of sorted disks to be used for partitioning
+        :rtype: list
+        """
+        for disk in req_disks:
+            if self.boot_disk and disk == self.boot_disk:
+                boot_index = req_disks.index(disk)
+                req_disks.insert(0, req_disks.pop(boot_index))
+
```

Comment:
Do not modify the list you are iterating over. I know this probably works in Python 3 somehow, but it's at least suspicious.